File: library/src/main/java/me/shkschneider/skeleton/ui/ScrollHelper.java
package me.shkschneider.skeleton.ui;

public class ScrollHelper {

    // <http://benjii.me/2010/08/endless-scrolling-listview-in-android/>
    public static boolean hasMoreItems(final int firstVisibleItem, final int visibleItemCount, final int totalItemCount, final int threshold) {
        return ((totalItemCount - visibleItemCount) <= (firstVisibleItem + threshold));
    }

}

File: library/src/main/java/me/shkschneider/skeleton/ui/ScrollHelper.kt
package me.shkschneider.skeleton.ui

object ScrollHelper {

    // <http://benjii.me/2010/08/endless-scrolling-listview-in-android/>
    fun hasMoreItems(firstVisibleItem: Int, visibleItemCount: Int, totalItemCount: Int, threshold: Int): Boolean {
        return totalItemCount - visibleItemCount <= firstVisibleItem + threshold
    }

}
